mne/viz/misc.py

```diff
@@ -625,3 +627,239 @@ def plot_dipole_amplitudes(dipoles, colors=None, show=True):
     return fig
+
+
+def adjust_axes(axes, remove_spines=('top', 'right'), grid=True):
+    """Adjust some properties of axes
+
+    Parameters
+    ----------
+    axes : list
+        List of axes to process.
+    remove_spines : list of str
+        Which axis spines to remove.
+    grid : bool
+        Turn grid on (True) or off (False).
+    """
+    axes = [axes] if not isinstance(axes, (list, tuple, np.ndarray)) else axes
+    for ax in axes:
+        if grid:
+            ax.grid(zorder=0)
+        for key in remove_spines:
+            ax.spines[key].set_visible(False)
+
+
+def _log_ticks(lims):
+    """Create approximately logarithmically spaced ticks between lims."""
+    lims = np.array(lims)
+    ticks = list()
+    for exp in range(int(np.floor(np.log10(lims[0]))),
+                     int(np.floor(np.log10(lims[1]))) + 1):
+        ticks += (np.array([1, 2, 4]) * (10 ** exp)).tolist()
+    ticks = np.array(ticks)
+    ticks = ticks[(ticks >= lims[0]) & (ticks <= lims[1])]
+    ticklabels = [('%g' if t < 1 else '%d') % t for t in ticks]
+    return ticks, ticklabels
+
+
+def _get_flim(flim, freq, sfreq=None):
+    """Get reasonable frequency limits."""
+    if flim is None:
+        if freq is None:
+            flim = [0.1, sfreq / 2.]
+        else:
+            flim = [freq[0] if freq[0] > 0 else 0.1 * freq[1], freq[-1]]
+    if flim[0] <= 0:
+        raise ValueError('flim[0] must be positive, got %s' % flim[0])
+    return flim
+
+
+def plot_filter(h, sfreq, freq=None, gain=None, title=None, color='#1f77b4',
+                flim=None, alim=(-60, 10), show=True):
+    """Plot properties of a filter
+
+    Parameters
+    ----------
+    h : dict or ndarray
+        Can be an IIR dict or 1D ndarray of coefficients (for FIR filter).
+    sfreq : float
+        Sample rate of the data (Hz).
+    freq : array-like or None
+        The ideal response frequencies to plot (must be in ascending order).
+        If None, do not plot the ideal response.
+    gain : array-like or None
+        The ideal response gains to plot.
+        If None, do not plot the ideal response.
+    title : str | None
+        The title to use. If None, deteremine the title based on the
+        type of the system.
+    color : color object
+        The color to use.
+    flim : tuple or None
+        If not None, the x-axis frequency limits (Hz) to use.
+        If None, freq will be used. If None and freq is None,
+        ``(0.1, sfreq / 2.)`` will be used.
+    alim : tuple
+        The y-axis amplitude limits (dB) to use.
+    show : bool
+        Show figure if True.
+
+    Returns
+    -------
+    fig : matplotlib.figure.Figure
+        The figure containing the plots.
+
+    Notes
+    -----
+    .. versionadded:: 0.13
+    """
+    from scipy.signal import freqz, group_delay
+    import matplotlib.pyplot as plt
+    sfreq = float(sfreq)
+    flim = _get_flim(flim, freq, sfreq)
+    omega = np.logspace(np.log10(flim[0]), np.log10(flim[1]), 1000)
```

Comment:
can the x log scale be optional (ie. possible to turn off with a param like `log_x`)? I sometimes like to look at the freq response with linear x.